src/app/views/recipe.ts

```diff
@@ -2,94 +2,81 @@ import $ from 'jquery';
 import * as moment from 'moment';
 
 import { getRecipe, getRecipeById } from '../common';
-import { db } from '../database';
+import { Recipe, Starred, db } from '../database';
 import { i18nAttr, localize } from '../i18n';
 import { addRecipe, scaleRecipe } from '../models/recipes';
 import { starRecipe, unstarRecipe } from '../models/starred';
-import { renderIngredientHTML, renderDirectionHTML } from '../recipeml';
+import { renderIngredientHTML } from '../recipeml';
 import { getState, loadPage, pushState, renderStateHash } from '../state';
 
 export { renderRecipe };
 
-function hoverDirection() {
-  $(this).addClass('hover');
-}
-
-function unhoverDirection() {
-  $(this).removeClass('hover');
-}
-
-function markDirection() {
-  $('#recipe div.directions li.direction').removeClass('mark');
-  $(this).addClass('mark');
-}
-
-function linkFormatter(recipe) {
+function linkFormatter(recipe: Recipe) : JQuery {
   return $('<a />', {
     'class': 'link fa fa-link',
     'href': `#search&action=view&id=${recipe.id}`
   });
 }
 
-function starFormatter() {
+function starFormatter() : JQuery {
```

Comment:
We seem a bit inconsistent about when we apply the `JQuery` return type.  Is this working as intended?